### PR TITLE
adding statusmonitor/main.go

### PR DESCRIPTION
**Pull request by vijayrap**
Raising a review request.

### CODE REVIEW

**Review by psykulsk — CHANGES_REQUESTED** [review]
Thanks @vijayrap, that's good, but more changes are required:

This function has to return error
https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/config_status_monitor.go#L56
Instead of doing log.Fatalf() here https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/config_status_monitor.go#L89

Same with the getControlStatus, it should return error:
https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/main.go#L118
instead of log.Fatalf: https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/main.go#L136
and here https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/main.go#L123

Then, in the main for loop, you can check those errors and log them here https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/main.go#L107 and here https://github.com/Juniper/contrail-operator/blob/new_2008R_update_CEM-18251/statusmonitor/main.go#L109 .
Thanks to that, they will be logged but the process will not exit.

File: statusmonitor/main.go [hunk 1]
```diff
@@ -93,16 +93,22 @@ func main() {
...
-					log.Fatalf("kubernates client creation failed: %v", err)
+					log.Printf("kubernates client creation failed: %v", err)
+					continue
 				}
 				ticker = time.NewTicker(time.Duration(config.Interval) * time.Second)
 				switch config.NodeType {
 				case "control":
-					getControlStatus(client, clientset, restClient, config)
+					err := getControlStatus(client, clientset, restClient, config)
+					if err != nil{
```
Comment (Dysproz) [inline statusmonitor/main.go:108]: It may be done as one liner:
```
if err := getControlStatus(client, clientset, restClient, config); err != nil {
```

**Review by Dysproz — APPROVED** [review]
LGTM. One NIT comment

**Review by psykulsk — APPROVED** [review]

**Comment by psykulsk** [comment]
recheck

**Pull request merged**